src/metrics_probes.erl

```diff
@@ -0,0 +1,144 @@
+%%==============================================================================
+%% Copyright 2019 Erlang Solutions Ltd.
+%%
+%% Licensed under the Apache License, Version 2.0 (the "License");
+%% you may not use this file except in compliance with the License.
+%% You may obtain a copy of the License at
+%%
+%% http://www.apache.org/licenses/LICENSE-2.0
+%%
+%% Unless required by applicable law or agreed to in writing, software
+%% distributed under the License is distributed on an "AS IS" BASIS,
+%% WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+%% See the License for the specific language governing permissions and
+%% limitations under the License.
+%%==============================================================================
+-module(metrics_probes).
+-behaviour(exometer_probe).
+
+-include("mongoose_logger.hrl").
+
+%% exometer_entry callbacks
+%% exometer_probe callbacks
+-export([
+         behaviour/0,
+         probe_init/3,
+         probe_terminate/1,
+         probe_get_value/2,
+         probe_get_datapoints/1,
+         probe_update/2,
+         probe_reset/1,
+         probe_sample/1,
+         probe_setopts/3,
+         probe_handle_msg/2,
+         probe_code_change/3
+        ]).
+
+%% Samples
+-export([
+         do_sample_queue_lengths/0,
```

Comment:
How about moving those functions to separate module in order to keep this one just a generic probe calling given MFA?

Reply:
I agree with this idea. Two questions: what name should I give to this module, and, should I think about moving the numerous modules related to metrics to a separate `metrics/` folder inside of `src/`? Just to organise the code a bit, or does it make no difference?

Reply:
What about `metrics_probes` and `metrics_probes_server`? And, yes, I all for moving all those into a sub-directory. `mongoose_metrics` is already very bloated and I would really like to see it split soon, so that would generate even more files.